gusto/diffusion.py

```diff
@@ -71,3 +74,64 @@ def apply(self, x_in, x_out):
         self.phi1.assign(x_in)
         self.solver.solve()
         x_out.assign(self.phi1)
+
+
+class RecoveredDiffusion(Diffusion):
+    """
+    A diffusion strategy for the lowest order spaces, allowing us to recover to a
+    higher order space, where diffusion will be performed.
+
+    :arg state: the state object.
+    :arg diffusion_scheme: an existing Diffusion class. For a vector field this should
```

Comment:
This seems a bit weird to me. I wonder if we should do recovered diffusion like we do recovered advection - i.e. with a decorator. Would be good to know if that is good coding or not...

Reply:
I think the decorator idea is definitely better than this -- I'll give that a go.